clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -118,6 +130,24 @@ public void assign(List<TopicPartition> partitions) {
         this.assignment.keySet().retainAll(this.userAssignment);
     }
 
+    public void subscribe(Pattern pattern) {
+        if (!this.subscription.isEmpty() || !this.userAssignment.isEmpty())
```

Comment:
Since subscribe(topics) is no longer additive, I wonder if we should have subscribe(pattern) just replace the previous topic subscription (if there was one) instead of raising the exception?